services/grpcService.ts

```diff
@@ -52,9 +54,94 @@ export class GrpcBlockchainClient implements BlockchainClient {
+    return (t: Transaction.AsObject, index: number, array: Transaction.AsObject[]): boolean => {
+      return (
+        (t.timestamp >= XEC_TIMESTAMP_THRESHOLD && address.networkId === XEC_NETWORK_ID) ||
+        (t.timestamp >= BCH_TIMESTAMP_THRESHOLD && address.networkId === BCH_NETWORK_ID)
+      )
+    }
+  }
+
+  private parseMempoolTx (mempoolTx: MempoolTransaction.AsObject): Transaction.AsObject {
+    const tx = mempoolTx.transaction!
+    tx.timestamp = mempoolTx.addedTime
+    return tx
+  }
+
+  private async getTransactionAmount (transaction: Transaction.AsObject, addressString: string): Promise<Prisma.Decimal> {
+    let totalOutput = 0
+    let totalInput = 0
+    const addressFormat = xecaddr.detectAddressFormat(addressString)
+    const unprefixedAddress = removeAddressPrefix(addressString)
+
+    for (const output of transaction.outputsList) {
+      let outAddress: string | undefined = removeAddressPrefix(output.address)
+      if (output.scriptClass === 'pubkey') {
+        outAddress = await pubkeyToAddress(outAddress, addressFormat)
+        if (outAddress !== undefined) outAddress = removeAddressPrefix(outAddress)
+      }
+      if (unprefixedAddress === outAddress) {
+        totalOutput += output.value
+      }
+    }
+    for (const input of transaction.inputsList) {
+      let addressFromPkey = await pubkeyToAddress(input.address, addressFormat)
+      if (addressFromPkey !== undefined) addressFromPkey = removeAddressPrefix(addressFromPkey)
+      if (unprefixedAddress === removeAddressPrefix(input.address) || unprefixedAddress === addressFromPkey) {
+        totalInput += input.value
+      }
+    }
+    const satoshis = new Prisma.Decimal(totalOutput).minus(totalInput)
+    return await satoshisToUnit(
+      satoshis,
+      addressFormat
+    )
+  }
+
+  private async getTransactionPrismaFromTransaction (transaction: Transaction.AsObject, addressString: string, confirmed: boolean): Promise<TransactionPrisma> {
+    return {
+      hash: transaction.hash as string,
+      amount: await this.getTransactionAmount(transaction, addressString),
+      timestamp: transaction.timestamp,
+      addressId: 0,
+      id: 0,
+      confirmed
+    }
+  }
+
+  public async getAddressTransactions (addressString: string, start: number = 0, maxTransactions: number = Infinity): Promise<TransactionsResponse> {
+    const address = await fetchAddressBySubstring(addressString)
+    const pageSize = FETCH_N
+    let newTransactionsCount = -1
+    let page = Math.floor(start / pageSize)
+    const confirmedTransactions: Transaction.AsObject[] = []
+    const unconfirmedTransactions: Transaction.AsObject[] = []
+
+    while (confirmedTransactions.length < maxTransactions && newTransactionsCount !== 0) {
+      const client = this.getClientForAddress(address.address)
+      const transactions = (await client.getAddressTransactions({
+        address: address.address,
+        nbSkip: page * pageSize,
```

Comment:
yep, but the page that we are in is calculated from the `start` parameter above (and it will work with any page size):

`let page = Math.floor(parameters.start / pageSize)`

so we start in the page of the last transaction where we left off.